docs/source/programming/SLURM-arguments.md

```diff
@@ -0,0 +1,147 @@
+(slurm-arguments-target)=
+# SLURM arguments primer
+
+```{include} ../_static/swc-wiki-warning.md
+```
+
+## Abbreviations
+| Acronym | Meaning                                      |
+| ------- | -------------------------------------------- |
+| SWC     | Sainsbury Wellcome Centre                    |
+| HPC     | High Performance Computing                   |
+| SLURM   | Simple Linux Utility for Resource Management |
+| MPI     | Message Passing Interface                    |
+
+
+## Overview
+SLURM is a job scheduler and resource manager used on the SWC HPC cluster.
+It is responsible for allocating resources (e.g. CPU cores, GPUs, memory) to jobs submitted by users.
+When submitting a job to SLURM, you can specify various arguments to request the resources you need.
+These are called SLURM directives, and they are passed to SLURM via the `sbatch` or `srun` commands.
+
+These are often specified at the top of a SLURM job script,
+e.g. the lines that start with `#SBATCH` in the following example:
+
+```{code-block} bash
+#!/bin/bash
+
+#SBATCH -J my_job # job name
+#SBATCH -p gpu # partition (queue)
+#SBATCH -N 1   # number of nodes
+#SBATCH --mem 16G # memory pool for all cores
+#SBATCH -n 4 # number of cores
+#SBATCH -t 0-06:00 # time (D-HH:MM)
+#SBATCH --gres gpu:1 # request 1 GPU (of any kind)
+#SBATCH -o slurm.%x.%N.%j.out # STDOUT
+#SBATCH -e slurm.%x.%N.%j.err # STDERR
+#SBATCH --mail-type=ALL
+#SBATCH --mail-user=user@domain.com
+#SBATCH --array=1-12%4 # job array index values
+
+# load modules
+...
+
+# execute commands
+...
+```
+This guide provides only a brief overview of the most important SLURM arguments,
+to demysify the above directives and help you get started with SLURM.
+For a more detailed description see the [SLURM documentation](https://slurm.schedmd.com/sbatch.html).
+
+##  Commonly used arguments
+
+### Partition (Queue)
+- *Name:* `--partition`
+- *Alias:* `-p`
+- *Description:* Specifies the partition (or queue) to submit the job to. To see a list of all partitions/queues, the nodes they contain and their respective time limits, type `sinfo` when logged in to the HPC cluster.
+- *Example values:* `gpu`, `cpu`, `fast`, `medium`
+
+### Job Name
+- *Name:* `--job-name`
+- *Alias:* `-J`
+- *Description:* Specifies a name for the job, which will appear in various SLURM commands and logs, making it easier to identify the job (especially when you have multiple jobs queued up)
+- *Example values:* `training_run_24`
+
+### Number of Nodes
+- *Name:* `--nodes`
+- *Alias:* `-N`
+- *Description:* Defines the number of nodes required for the job.
+- *Example values:* `1`
+
+:::{warning}
+This should always be `1`, unless you really know what you're doing,
+e.g. you are parallelising your code across multiple nodes with MPI.
+:::
+
+### Number of Cores
+- *Name:* `--ntasks`
+- *Alias:* `-n`
+- *Description:* Defines the number of cores (or tasks) required for the job.
+- *Example values:* `4`, `8`, `16`
```

Comment:
Just to make it clear that `1` is often adequate, and that they don't need to be powers of 2 (as lots of things do need to be in computing).
```suggestion
- *Example values:* `1`, `5`, `20`
```